sql/database.go

```diff
@@ -223,8 +224,12 @@ type Opt func(c *conf)
 
 // OpenInMemory creates an in-memory database.
 func OpenInMemory(opts ...Opt) (*sqliteDatabase, error) {
-	opts = append(opts, WithConnections(1), withForceFresh())
-	return Open("file::memory:?mode=memory", opts...)
+	opts = append(opts, withForceFresh())
+	// Unique uri is needed to avoid sharing the same in-memory database,
+	// while allowing multiple connections to the same database.
+	uri := fmt.Sprintf("file:mem-%d-%d?mode=memory&cache=shared",
+		rand.Uint64(), rand.Uint64())
```

Comment:
NIT: one `rand.Uint64()` should be enough to make collisions impossible 🙂

Reply:
Fixed